Mathlib/Tactic/Linter/TextBased.lean

```diff
@@ -213,8 +219,9 @@ def formatErrors (errors : Array ErrorContext) (style : ErrorFormat) : IO Unit :
     IO.println (outputMessage e style)
 
 /-- Core logic of a text based linter: given a collection of lines,
-return an array of all style errors with line numbers. -/
-abbrev TextbasedLinter := Array String → Array (StyleError × ℕ)
+return an array of all style errors with line numbers and, optionally, an array of lines
+changed to fix the linter automatically. (This is not always possible, but sometimes it is.) -/
```

Comment:
The new optional array being returned... is this all the lines of the file? Just lines that were changed? If so, which? Hopefully we just need a better doc-string here.

Reply:
If it's just the whole file, would it made sense to intercalate all the lines with `\n` and just return a single string?

Reply:
It's all the lines of the file. Is the new docstring clear enough?

It seems tempting, but the answer is no: if there is more than one auto-fixable text-based linter (and there will be in the future), one would like the auto-fixes to compound --- fixing first one error, then the other etc. In particular, later linters need to work with the result of auto-fixing, thus need an array of lines to work with. Intercalating right away would not help, as one would only need to split the string again.